core/src/main/java/org/radargun/stats/representation/DataThroughput.java

```diff
@@ -28,24 +32,24 @@ public static DataThroughput compute(long totalBytes, long[] responseTimes, int
       double maxDurationSecs = Long.MIN_VALUE;
       double meanDurationSecs = 0;
       double[] throughputs = new double[requestCount];
+      long[] requestTimes = new long[requestCount];
 
       if (requestCount == 0) {
-         return new DataThroughput(0, 0, 0, 0);
+         return new DataThroughput(0, 0, 0, 0, 0, new long[0]);
       } else {
          for (int i = 0; i < requestCount; ++i) {
-            minDurationSecs = Math.min(minDurationSecs, toSecs(responseTimes[i]));
-            maxDurationSecs = Math.max(maxDurationSecs, toSecs(responseTimes[i]));
+            minDurationSecs = Math.min(minDurationSecs, responseTimes[i]);
+            maxDurationSecs = Math.max(maxDurationSecs, responseTimes[i]);
             meanDurationSecs += responseTimes[i];
             throughputs[i] = totalBytes / toSecs(responseTimes[i]);
+            requestTimes[i] = responseTimes[i];
          }
+         minDurationSecs = toSecs(minDurationSecs);
+         maxDurationSecs = toSecs(maxDurationSecs);
          meanDurationSecs = toSecs(meanDurationSecs / requestCount);
       }
       return new DataThroughput(totalBytes / maxDurationSecs, totalBytes / minDurationSecs, totalBytes
-            / meanDurationSecs, getDeviation(throughputs));
-   }
-
-   private static double toSecs(long nanos) {
-      return nanos / TimeUnit.SECONDS.toNanos(1);
```

Comment:
Nanoseconds are converted to seconds rather with TimeUnit.NANOSECONDS.toSeconds(nanos) :)